### DISCUSSION TITLE
HowTo "import as" for pupdevices

### DISCUSSION THREAD

**Discussion by hwillemen** [pybricks/support#976] [Q&A]
I want to import all the classes in pybricks.pupdevices
I do not want have a huge list cluttering my file though.

"from pybricks.pupdevices import *" works but the linter complains that it does not know where the imported classes come from.

Normally in python I would be able to:
"import module.submodule as short" and have a short name reference to the library.
When I do this in pybricks I get:
AttributeError: 'module' object has no attribute 'pupdevices'

So how can I fix this?

PS: I even tried a weird structure like "from pybricks.pupdevices import * as pupdev"

**Comment by laurensvalk** [comment]
This is not currently supported, although it would be possible to support it.

This is typically the case with MicroPython/Pybricks. In this case, there are already 2 valid ways to do it, so we did not feel compelled to add a 3rd way to do it, to save space.

But likewise, if it turns out that people ask about those features, we can certainly consider adding them.

**Comment by hwillemen** [comment]
Is it a lot of effort to support the "as" keyword?
Feels as basic python to me since it just creates a reference and should not take up any space in the program.

The first 2 ways create a lot of clutter in the code:
1) from pybricks.pupdevices import <long list of classes>
2) import pybricks.pupdevices
   
for each class usage:
pybricks.pupdevices.<class>

The 3rd way is allowed by the IDE but not by the linter when I want to add a script to the repo:
(and I understand why)
3) from pybricks.pupdevices import *

thank you in advance.

**Reply by laurensvalk** [reply]
Using `as` is supported.

Only importing modules that way isn't. I'll have a go at implementing it to see the required size.
